dianna/visualization/text.py

```diff
@@ -1,69 +1,101 @@
-        else:
-            # normalize to max importance
-            importance = importance / max_importance
-            color = _get_color(importance, opacity)
-
-        tag = (f'<mark style="background-color: {color}; '
-               f'line-height:1.75">{token}</mark>')
-        tags.append(tag)
-
-    html = ' '.join(tags)
-
-    return html
-
-
-def _get_color(importance: float, opacity: float) -> str:
-    # clip values to prevent CSS errors (Values should be from [-1,1])
-    importance = max(-1, min(1, importance))
-    if importance > 0:
-        hue = 0
-        sat = 100
-        lig = 100 - int(50 * importance)
-    else:
-        hue = 240
-        sat = 100
-        lig = 100 - int(-50 * importance)
-    return f'hsl({hue}, {sat}%, {lig}%, {opacity})'
+    tokens, _, importances = zip(*explanation)
+
+    if input_tokens:
+        # Make a list of tuples (token, i, importance) for each token in the
+        # input_tokens. if a token isnot in the explanantion, the importance is
+        # None
+        explanation = [
+            (
+                token, i, importances[tokens.index(token)] if token in tokens else None
+            )
+            for i, token in enumerate(input_tokens)
+        ]
+
+    vmin, vmax = heatmap_range
+
+    x, y = (0, 0) # the initial position of the text
+    space_token = ' '
+    fig, ax = plt.subplots(figsize=(10, 1))
+    ax.axis('off')
+
+    for token, _, importance in explanation:
+
+        color = _get_text_color(importance, vmin, vmax, colormap, alpha)
+        text = ax.text(x, y, token, fontsize=12, backgroundcolor=color)
+
+        # Get the bounding box of the text in display space and convert it to data space
+        bbox = text.get_window_extent()
+        bbox_data = mtransforms.Bbox(ax.transData.inverted().transform(bbox))
+        x = bbox_data.x1
+
+        # Add a space after each token
+        text = ax.text(x, y, space_token, fontsize=12)
+        bbox = text.get_window_extent()
+        bbox_data = mtransforms.Bbox(ax.transData.inverted().transform(bbox))
+
+        # The next x is the right side of the bbox plus the fixed space
+        x = bbox_data.x1
+
+        # Wrap the text if token is a dot
+        if token == '.':
+            x = 0
+            y -= 0.5  # space between lines in inches
+
+    # adjust the height of the figure
+    y_hight = 1
+    if abs(y) > 1:
+        y_hight = y
+    ax.set_ylim(y_hight, 0)
+    fig.set_figheight(abs(y_hight))
+
+    # add colorbar
+    sm = plt.cm.ScalarMappable(
+        cmap=plt.get_cmap(colormap),
+        norm=plt.Normalize(vmin, vmax)
+        )
+    sm.set_array([])
+    fig.colorbar(sm, ax=ax, orientation='horizontal', aspect=20, use_gridspec=True)
+    # TODO add alpha to the colorbar
+
+    if not show_plot:
+        plt.close()
+
+    if output_filename:
+        plt.savefig(output_filename)
+
+    return fig, ax
+
+
+def _get_text_color(importance, vmin, vmax, colormap, alpha):
+    if importance is None:
```

Comment:
An inline comment on what `importance` is would help here. It's possible to retrieve, but one needs to look the the parent function's body. Not suggesting a full parameter description as that would be redundant.

Reply:
I added docstring to this function.